### PR TITLE
CORCI-666 build: Better handling of WebRetriever downloads

### PR DESCRIPTION
**Pull request by jolivier23**
Specify a checksum and only download if there is no existing
file that matches.  Also, retry the download a few times on
failure to avoid issues with intermittent failures.
Make minor changes to fuse build to ensure we don't try to build master

### CODE REVIEW

File: prereq_tools/base.py [hunk 1]
```diff
@@ -487,26 +487,61 @@ def update(self, subdir, **kw):
...
+
+    def download(self, basename):
+        """Download the file"""
+        # Retry download a few times if it fails
+        for i in range(0, 3):
+            command = ['curl -L -O %s' % self.url]
+
+            if not RUNNER.run_commands(command):
+                print "Try #%d to get %s failed" % (i + 1, self.url)
+                continue
```
Comment (brianjmurrell) [inline prereq_tools/base.py]: Maybe a sleep in here to try to avoid doing all three attempts during a short outage?

Comment (jolivier23) [reply prereq_tools/base.py]: yeah, I thought of that but was lazy.  I just added a simple sleep with exponential backoff

File: prereq_tools/base.py [hunk 2]
```diff
@@ -530,6 +533,10 @@ def download(self, basename):
 
             print "Try #%d to get %s failed due to md5 mismatch" % (i + 1,
                                                                     self.url)
+            if (i + 1) != tries:
```
Comment (brianjmurrell) [inline prereq_tools/base.py]: Don't we want this sleep to run in the line 526 failure case also?

Comment (jolivier23) [reply prereq_tools/base.py]: good point

File: prereq_tools/base.py (hunk 1 above)
Comment (jolivier23) [reply prereq_tools/base.py]: I rearranged it to do backoff in both cases

File: Jenkinsfile [hunk 3]
```diff
@@ -174,14 +174,15 @@ pipeline {
             // Newer one looked up the last known good build of master branch.
             // Pipeline currently does not have access that info.
             sh 'rm -rf testbin/ && mkdir -p testbin'
+            sh 'rm -rf srcpath/fuse'
```
Comment (ashleypittman) [inline Jenkinsfile]: Is this still needed?

Comment (jolivier23) [reply Jenkinsfile]: Maybe not but it doesn’t hurt anything.  Just making sure the slate is clean

**Pull request merged**